docs/tasks/administer-cluster/dns-custom-nameservers.md

```diff
@@ -55,66 +39,82 @@ data:
-stub domain suffix (e.g. “.acme.local”) are sent to the configured custom
-resolver. Finally, requests that do not match any of those suffixes are
-forwarded to the upstream DNS.
+ As specified, DNS requests with the “.acme.local” suffix
+are forwarded to a DNS listening at 1.2.3.4. Google Public DNS
+serves the upstream queries.
 
-Below is a table of example domain names and the destination of the queries for
-those domain names:
+The table below describes how queries with certain domain names would map to their destination DNS servers:
 
 | Domain name | Server answering the query |
 | ----------- | -------------------------- |
 | kubernetes.default.svc.cluster.local| kube-dns |
 | foo.acme.local| custom DNS (1.2.3.4) |
 | widget.com    | upstream DNS (one of 8.8.8.8, 8.8.4.4) |
 
+See [ConfigMap options](#configmap-options) for
+details about the configuration option format.
+
 {% endcapture %}
 
 {% capture discussion %}
 
-## Understanding custom DNS upstream servers and stub domains
-
-### Pod DNS policies
+## Understanding name resolution in Kubernetes
 
 Kubernetes currently supports two DNS policies specified on a per-pod basis
-using the dnsPolicy flag: “Default” and “ClusterFirst”. If dnsPolicy is not
-explicitly specified, then “ClusterFirst” is used:
+using the `dnsPolicy` flag: “Default” and “ClusterFirst”.
+
+*NOTE: "Default" is not the default DNS policy. If `dnsPolicy` is not
+explicitly specified, then “ClusterFirst” is used.*
+
+### "Default" DNS Policy
+
+If `dnsPolicy` is set to “Default”, then the name resolution configuration is
+inherited from the node that the pods run on.
+
+Custom upstream nameservers and
+stub domains cannot be used in conjunction with this DNS policy.
 
-If dnsPolicy is set to “Default”, then the name resolution configuration is
-inherited from the node the pods run on. Note: custom upstream nameservers and
-stub domains cannot be used in conjunction with dnsPolicy: “Default”.
 
-If dnsPolicy is set to “ClusterFirst”, then DNS queries are sent to the
-kube-dns service. Queries for domains rooted in the configured cluster domain
-suffix (any address ending in “.cluster.local” in the example above) are
-answered by the kube-dns service. All other queries, such as
-www.kubernetes.io, are forwarded to the upstream nameserver inherited from
-the node.
+### "ClusterFirst" DNS Policy
+
+If the `dnsPolicy` is set to "ClusterFirst", name resolution is handled differently, *depending on whether stub-domain and upstream DNS servers are configured*.
+
+If they are *not* configured, any query that does not match the configured cluster domain suffix, such as "www.kubernetes.io", is forwarded to the upstream nameserver inherited from the node.
+
+If stub domains and upstream DNS servers are configured (as in the [previous example](#configuring-stub-domain-and-upstream-dns-servers)), DNS queries will be
+routed according to the following flow:
+1. The query is first sent to the DNS caching layer in kube-dns.
```

Comment:
Looks like this needs a newline above it so that the ordered list renders correctly.